ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/dropdownActivityPicklistQuestion.component.ts

```diff
@@ -9,54 +9,60 @@ import { NGXTranslateService } from '../../../services/internationalization/ngxT
-                   [placeholder]="details.placeholder"
-                   [value]="details.text">
-        </mat-form-field>
-        <p *ngIf="!readonly" class="ddp-helper">
-            <span class="ddp-counter-color">
-                {{ questionIdToCharactersLeft[details.stableId] }}
-            </span>{{ questionIdToCharacterLeftMsg[details.stableId] }}
-        </p>
-    </ng-container>`,
+    <mat-form-field [floatLabel]="block.picklistLabel && block.selectMode === SELECT_MODE.SINGLE ? 'always' : null">
+      <mat-label *ngIf="block.picklistLabel && block.selectMode === SELECT_MODE.SINGLE">
+        {{block.picklistLabel}}
+      </mat-label>
+
+      <ng-container *ngIf="block.selectMode === SELECT_MODE.MULTIPLE; then matSelect else nativeSelect">
+      </ng-container>
+
+      <ng-template #matSelect>
+          <div class="ddp-dropdown">
+              <mat-select [value]="setMaterialSelected()"
+                          [placeholder]="block.picklistLabel"
+                          [disabled]="readonly"
+                          [multiple]="true"
+                          (selectionChange)="handleMaterialSelect($event); details.show ? updateCharactersLeftIndicator(details.stableId) : null"
+                          class="width">
+                  <mat-option *ngFor="let option of block.picklistOptions"
+                              [value]="option.stableId">
+                      {{option.optionLabel}}
+                  </mat-option>
+              </mat-select>
+          </div>
+      </ng-template>
+
+      <ng-template #nativeSelect>
+          <select matNativeControl
+                  [(ngModel)]="nativeSelectedValue"
+                  [disabled]="readonly"
+                  (change)="handleNativeSelect($event.target.value); details.show ? updateCharactersLeftIndicator(details.stableId) : null"
+                  class="width">
+              <option *ngFor="let option of block.picklistOptions"
+                      [value]="option.stableId">
+                  {{option.optionLabel}}
+              </option>
+          </select>
+      </ng-template>
+
+      <ng-container *ngIf="details.show">
+          <mat-form-field class="width ddp-option-details-field">
+              <input matInput
+                     (change)="detailTextChanged($event.target.value)"
+                     (input)="updateCharactersLeftIndicator(details.stableId, $event.target.value)"
+                     [disabled]="readonly"
+                     [attr.maxlength]="block.detailMaxLength"
+                     [placeholder]="details.placeholder"
+                     [value]="details.text">
+          </mat-form-field>
+          <p *ngIf="!readonly" class="ddp-helper">
+              <span class="ddp-counter-color">
+                  {{ questionIdToCharactersLeft[details.stableId] }}
+              </span>{{ questionIdToCharacterLeftMsg[details.stableId] }}
+          </p>
+      </ng-container>
+    </mat-form-field>`,
```

Comment:
@aglushko-broad : unfortunately git diff is not really helping to narrow down the change you made here. Could you briefly highlight/explain?

Reply:
sure! the changes I made:
- wrap with mat-form-field and assign matNativeControl to native select in order to apply material styles for alignment
- remove the first option from native select which was equal to the label and add mat-label (I had to add it out of native select ng-template in order to let internal mechanism which uses ng-content to handle it)